pkgs/development/python-modules/textx/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, arpeggio
+, click
+, coverage
+, jinja2
+, py
+, pytest
+, python
+, setuptools
+}:
+
+buildPythonPackage rec {
+  pname = "textX";
+  version = "2.2.0";
+
+  src = fetchPypi {
+    inherit pname;
+    inherit version;
+    sha256 = "00mwd588ms96qp27m5vpjkzk30wfw53hnmv8y77slxca8lw9vq82";
+  };
+
+  propagatedBuildInputs = [
+    arpeggio
+    click
+    setuptools
+  ];
+
+  checkInputs = [
+    coverage
+    jinja2
+    py
+    pytest
+  ];
+
+  checkPhase = ''
+     runHook preCheck
+
+     ${python.interpreter} -m py.test tests/functional
+
+     runHook postCheck
```

Comment:
`pytestCheckHook` in `nativeBuildInputs` and than set `pytestFlagsArray = [ "tests/functional" ];`